python/BioSimSpace/FreeEnergy/_relative.py

```diff
@@ -417,7 +675,292 @@ def _analyse(self):
+               standard error.
+
+           overlap : [ [ float, float, ... ] ]
+               The overlap matrix. This gives the overlap between each lambda
+               window.
+        """
+
+        function_glob_dict = {
+            "SOMD": (Relative._somd_extract_u_nk),
+            "GROMACS": (_gmx_extract_u_nk),
+            "AMBER": (_amber_extract_u_nk)
+        }
+
+        # Extract the data.
+        func = function_glob_dict[engine]
+        try:
+            u_nk = [func(x, T=t) for x, t in zip(files, temperatures)]
+        except:
+            raise _AnalysisError(
+                "Could not extract the data from the provided files!")
+
+        # Preprocess the data.
+        try:
+            processed_u_nk = Relative.preprocessing_extracted_data(u_nk)
+        except:
+            _warnings.warn("Could not preprocess the data.")
+            processed_u_nk = u_nk
+
+        try:
+            mbar = _AutoMBAR().fit(processed_u_nk)
+        except:
+            raise _AnalysisError("MBAR free-energy analysis failed!")
+
+        # Extract the data from the mbar results.
+        data = []
+        # Convert the data frames to kcal/mol.
+        delta_f_ = _to_kcalmol(mbar.delta_f_)
+        d_delta_f_ = _to_kcalmol(mbar.d_delta_f_)
+        for lambda_, t in zip(lambdas, temperatures):
+            x = lambdas.index(lambda_)
+            mbar_value = delta_f_.iloc[0, x]
+            mbar_error = d_delta_f_.iloc[1, x]
+
+            # Append the data.
+            data.append((lambda_,
+                        (mbar_value) * _Units.Energy.kcal_per_mol,
+                        (mbar_error) * _Units.Energy.kcal_per_mol))
+
+        # Calculate overlap matrix.
+        overlap = mbar.overlap_matrix
+
+        return (data, overlap)
+
+    @staticmethod
+    def analyse_ti(files, temperatures, lambdas, engine):
+        """Analyse existing free-energy data using TI and the alchemlyb library.
+
+           Parameters
+           ----------
+
+           files : list
+               List of files for all lambda values to analyse. Should be sorted.
+
+           temperatures : list
+               List of temperatures at which the simulation was carried out at for each lambda window.
+               Index of the temperature value should match it's corresponding lambda window index in files.
+
+           lambdas : list
+               Sorted list of lambda values used for the simulation.
+
+           engine : str
+               Engine with which the simulation was run.
+
+           Returns
+           -------
+
+           pmf : [(float, :class:`Energy <BioSimSpace.Types.Energy>`, :class:`Energy <BioSimSpace.Types.Energy>`)]
+               The potential of mean force (PMF). The data is a list of tuples,
+               where each tuple contains the lambda value, the PMF, and the
+               standard error.
+
+           overlap : [ [ float, float, ... ] ]
```

Comment:
I guess `overlap` doesn't make sense here, so we have some inconsistency in what is returned depending on what estimator is used. Is the TI gradient object the same, i.e. a matrix of floats, or different?

It might be worth thinking about having separate `analyseMBAR` and `analyseTI` methods, rather than getting different output based on the estimator. As things stand, we would need to update _all_ docstrings to make it clear that what is returned in each case.

Reply:
I've tried to make it a bit more clear for now by changing it to dHdl (and also put the correct type which is `alchemlyb.estimators.ti_.TI`), but I agree and I think it might be cleaner to have different analyse for each. Also, currently the estimator is set during the init so that if an object has been run it is still able to be self analysed as before. However, this also currently limits this self analyse to only one estimator (unless that property is modified before analysing with the other estimator, but this doesn't feel very intuitive).  I haven't really been using it like this as I usually have run any perturbations outside of BSS and therefore can just choose which estimator to use when doing the full `BSS.FreeEnergy.Relative.analyse()` , but yea I think having seperate analysers for each might be a good idea as it would also make this part more clear.

Reply:
Yes, there's probably no need to specify the estimator when the `FreeEnergy.Relative` object is initialised, rather have it solely as an option to the `analyse` method (with a default). I'll have a think about separate analysers, since I agree that it is the clearer solution. Perhaps we could have, e.g. `analyseMBAR` and `analyseTI`, with the regular `analyse` method simply calling those, and left in place for backwards compatibility.